skrub/_text_encoder.py

```diff
@@ -0,0 +1,320 @@
+          with that name.
+
+        The following models have a good performance/memory usage tradeoff:
+
+        - ``intfloat/e5-small-v2``
+        - ``all-MiniLM-L6-v2``
+        - ``all-mpnet-base-v2``
+
+        You can find more options on the `sentence-transformers documentation
+        <https://www.sbert.net/docs/pretrained_models.html#model-overview>`_.
+
+        The default model is a shrinked version of e5-v2, which has shown good
+        performance in the benchmark of [1]_.
+
+    n_components : int or None, default=30,
+        The number of embedding dimensions. As the number of dimensions is different
+        accross embedding models, this class uses a :class:`~sklearn.decomposition.PCA`
+        to set the number of embedding to ``n_components`` during ``transform``.
+        Set ``n_components=None`` to skip the PCA dimension reduction mechanism.
+
+        See [1]_ for more details on the choice of the PCA and default ``n_components``.
+
+    device : str, default=None
+        Device (e.g. "cpu", "cuda", "mps") that should be used for computation. If None,
+        checks if a GPU can be used.
+        Note that macOS ARM64 users can enable the GPU on their local machine
+        by setting ``device="mps"``.
+
+    batch_size : int, default=32
+        The batch size to use during ``transform``.
+
+    use_auth_token : bool or str, default=None
+        HuggingFace authentication token to download private models.
+
+    cache_folder : str, default=None
+        Path to store models. By default ``~/skrub_data``.
+        See :func:`skrub.datasets._utils.get_data_dir`.
+
+    random_state : int, RandomState instance or None, default=None
+        Used when the PCA dimension reduction mechanism is used, for reproducible
+        results across multiple function calls.
+
+    verbose : bool, default=True
+        Verbose level, controls whether to show a progress bar or not during
+        ``transform``.
+
+    References
+    ----------
+    .. [1]  L. Grinsztajn, M. Kim, E. Oyallon, G. Varoquaux
+            "Vectorizing string entries for data processing on tables: when are larger
+            language models better?", 2023.
+            https://hal.science/hal-04345931
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> from skrub import TextEncoder
+
+    Let's encode video comments using only 2 embedding dimensions:
+
+    >>> enc = TextEncoder(model_name='intfloat/e5-small-v2', n_components=2)
+    >>> X = pd.Series([
+    ...   "The professor snatched a good interview out of the jaws of these questions.",
+    ...   "Bookmarking this to watch later.",
+    ...   "When you don't know the lyrics of the song except the chorus",
+    ... ], name='video comments')
+
+    Fitting does not train the underlying pre-trained deep-learning model,
+    but ensure various checks and enable dimension reduction.
+
+    >>> enc.fit_transform(X)
```

Comment:
```suggestion
    >>> enc.fit_transform(X) # doctest: +SKIP
```

?

so we can run pytest without transformers installed